src/main/kotlin/com/faire/detekt/rules/DoNotAssertIsEqualOnTheResultOfSingle.kt

```diff
@@ -0,0 +1,80 @@
+package com.faire.detekt.rules
+
+import com.faire.detekt.utils.isAssertThat
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.psi.psiUtil.astReplace
+import org.jetbrains.kotlin.psi.psiUtil.referenceExpression
+
+/**
+ * Do not assert isEqual on the result of single(). Instead, use containsOnly.
```

Comment:
Let's add the reason why this rule exists: using `.single()` will throw an exception if the collection does not contain exactly one element. The exception results in the assertion failure message being hidden, which makes debugging more difficult.

Reply:
Added that word by word to the class comment.